EMAGCacheBundle.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace eMAG\CacheBundle;
+
+use eMAG\CacheBundle\DependencyInjection\Compiler\CacheCompilerPass;
+use eMAG\CacheBundle\DependencyInjection\EMAGCacheExtension;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\HttpKernel\Bundle\Bundle;
+
+class EMAGCacheBundle extends Bundle
+{
+    protected $autoloader;
+
+    /**
+     * @inheritDoc
+     */
+    public function build(ContainerBuilder $container)
+    {
+        parent::build($container);
+
+        $container->addCompilerPass(new CacheCompilerPass());
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function boot()
+    {
+        $this->autoloader = spl_autoload_register($this->container->get('emag.cache.proxy.config')->getProxyAutoloader());
+    }
```

Comment:
spl_autoload_register return bool. Not a reference to the autoloader function. I don't think the shutdown is really required.